QuickFIXn/SessionSchedule.cs

```diff
@@ -183,6 +197,18 @@ private bool CheckTime(TimeSpan time)
             return true;
         }
 
+        private bool CheckWeekDay(System.DateTime time)
+        {
+            if (NonStopSession)
+            {
+                throw new InvalidOperationException("NonStopSession is set -- this should never be called.");
+            }
+
+            if (_weekdays.Contains(time.DayOfWeek))
```

Comment:
This logic is faulty.  If midnight falls within the session (e.g. it goes from 8am-2am), then it will erroneously return false between midnight-2am.

I'm fixing it in a new branch and will submit a new PR.